src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -2442,11 +2442,18 @@ class NodeUtils(object):
             val -= val % (1024 * 1024)  # round down to MB
             host_resc_avail['hpmem'] = \
                 pbs.size(convert_size(val, 'mb'))
+            if (self.cfg['cgroup']['memsw']['enabled']
+                    and self.cfg['cgroup']['memsw']['manage_cgswap']):
+                host_resc_avail['cgswap'] = \
+                    host_resc_avail['vmem']-host_resc_avail['mem']
```

Comment:
Should we guard against what if host_resc_avail['vmem'] - host_resc_avail['mem'] yield a negative value?

Reply:
It is (not strictly) positive by construction. The difference is the swap not reserved for other purposes, in this snippet in get_vmem_on_node:

`        if swap <= 0:
            pbs.logmsg(pbs.EVENT_DEBUG4,
                       '%s: No swap space detected' %
                       caller_name())
            swap = 0
        pbs.logmsg(pbs.EVENT_DEBUG4, 'total swap: %d' % swap)
        # Calculate reserved swap
        reserved = 0
        if not ignore_reserved:
            reserve_pct = int(self.cfg['cgroup']['memsw']['reserve_percent'])
            reserved += int(swap * (reserve_pct / 100.0))
            reserve_amount = self.cfg['cgroup']['memsw']['reserve_amount']
            reserved += size_as_int(reserve_amount)
            pbs.logmsg(pbs.EVENT_DEBUG4, 'reserved swap: %d' % reserved)
            if reserved > swap:
                reserved = swap
        # Calculate remaining vmem
        remaining = total + swap - reserved
` 
That last guard will guarantee that swap - reserved is always positive and that vmem remaining reported is at least mem.

But of course, if you'd rather see us add a max(0,...) I'm not opposed to it.

Reply:
In fact, given this is only done in exechost_startup, I'll add the max. No point in making a reader even think about that possibility...